sql: skip dropped constraints in introspection queries

spilchen opened the pull request:
Dropped constraints remain in table descriptors until the schema change job fully completes. During this time, introspection queries (such as crdb_internal.create_statements) may encounter these constraints, even if they reference columns that are also in the process of being dropped. This can result in errors.

This change updates the introspection logic to ignore constraints that are marked for deletion, preventing such failures.

Informs #147514
Epic: None

Release note (bug fix): Fixed a bug where introspection queries could fail if a dropped constraint referenced a column that was also being dropped.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/147773)
<!-- Reviewable:end -->

spilchen committed:
sql: skip dropped constraints in introspection queries

Dropped constraints remain in table descriptors until the schema change
job fully completes. During this time, introspection queries (such as
crdb_internal.create_statements) may encounter these constraints, even
if they reference columns that are also in the process of being dropped.
This can result in errors.

This change updates the introspection logic to ignore constraints that
are marked for deletion, preventing such failures.

Informs #147514
Epic: None

Release note (bug fix): Fixed a bug where introspection queries could
fail if a dropped constraint referenced a column that was also being
dropped.

fqazi reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> Nice find!

> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/147773#-:-ORw0lPjC9rpGcPasY5T:bcdfsno)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @spilchen)



<!-- Sent from Reviewable.io -->

spilchen commented:
TFTR!

bors r+
